pymongo/asynchronous/mongo_client.py

```diff
@@ -1173,6 +1175,30 @@ def __getitem__(self, name: str) -> database.AsyncDatabase[_DocumentType]:
         """
         return database.AsyncDatabase(self, name)
 
+    def __del__(self) -> None:
+        """Check that this AsyncMongoClient has been closed and issue a warning if not."""
+        # TODO: Remove in https://jira.mongodb.org/browse/PYTHON-4731
+        try:
+            if not self._closed:
+                if _IS_SYNC:
```

Comment:
I'd prefer to remove the `await` from the error message to reduce the duplication here.